8.x/theme-development/client-side-cart-module.md

```diff
@@ -0,0 +1,52 @@
+## Client Side cart module
+
+[[toc]]
+
+The Larammerce Template Engine has the **local card module**. In this tutorial, the source code of this module is Reviewed.
+
+**local card module** is in the following directory `resources/assets/js/define/local_cart_service.js`.
+It has 437 lines of code and is based on require js but there is no requirement to use require js. To read more about it, refer to this address:[requirejs](https://www.requirejs.org/).
+
+Let's check the local card module:
+
+requirejs has two very important functions called define and require. In this module, there is a define function that takes three inputs:
+
+The first input is the name of the module, the second input is an arraylist of the dependencies of this module
+and the third input of the function body module.
+
+```js
+define('local_cart_service', ['jquery', 'jq_cookie', 'tools', 'template', 'underscore'],
+    function (jQuery, cookie, tools, template, _) {
+```
+
+In the next part, there are constants of the module.
+
+**cartCountEl** constant shows the number of products in the cart with the cart-count selector.
+```js
+ const cartCountEl = jQuery('.cart-count');
+```
+
+In the box where the product content is presented and the operations of adding to the shopping cart and removing from the shopping cart are performed, the **product-box** attribute must be given to its HTML tag.
+```js
+   const productSelector = '[product-box]';
+```
+
+This constant is the name of the cookie's key in which the shopping cart information is located.
+```js
+const cartCookie = window.siteEnv.SITE_LOCAL_CART_COOKIE_NAME;
+```
```

Comment:
Here should be a description of the window.siteEnv object. So that the reader can understand more details about the system architecture. Something like this:

The `window.siteEnv` object is provided by the backend system consisting of the environment variables placed in the `.env` file, starting with `SITE_`, managed by the system administrator so that the front-end programmer can have access to this kind of configuration to create a more dynamic code structure.